pkg/cephmgr/rgw/bucket.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2017 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package rgw
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/model"
+)
+
+type rgwBucketStats struct {
+	Bucket string `json:"bucket"`
+	Usage  map[string]struct {
+		Size            uint64 `json:"size"`
+		NumberOfObjects uint64 `json:"num_objects"`
+	}
+}
+
+func GetBucketStats(context *clusterd.Context, runner AdminCommandRunner) (map[string]model.ObjectBucketStats, error) {
+	logger.Infof("Listing buckets")
+
+	result, err := runner.RunAdminCommand(context,
+		"bucket",
+		"stats")
+	if err != nil {
+		return nil, fmt.Errorf("failed to list buckets: %+v", err)
+	}
+
+	var rgwStats []rgwBucketStats
+	if err := json.Unmarshal([]byte(result), &rgwStats); err != nil {
+		return nil, fmt.Errorf("failed to read buckets stats. %+v, result=%s", err, result)
+	}
+
+	stats := map[string]model.ObjectBucketStats{}
+
+	for _, rgwStat := range rgwStats {
+		stat := model.ObjectBucketStats{Size: 0, NumberOfObjects: 0}
+		for _, usage := range rgwStat.Usage {
+			stat.Size = stat.Size + usage.Size
```

Comment:
The usage command returns usage per pool so I'm just summing them up.  The stat is just to hold the final result of summing all Usage.